### PR TITLE
Clean up initilization system of partials. handlebarsHelpers and handlebarsPartials are modules that need to be called handlebarsHelpers(HandleBars)

### PR DESCRIPTION
**Pull request by duttonw**
Copy all handlebars to components since it references handlebar partials which can't work when in assets/js.
handlebars.helpers.js still in same place during transition but handlebars are all in the components parts.

dist/components/handlebars.init.min.js is the bundled full version.

### CODE REVIEW

File: .esbuild/plugins/qgds-plugin-copy-assets.js [hunk 1]
```diff
@@ -18,13 +18,14 @@ export default function copyPlugin() {
         to: ["./dist/sample-data/"],
       },
       {
-        from: ["./src/js/handlebars*"],
-        to: ["./dist/assets/js"],
+        from: ["./src/js/handlebars.helpers.js"],
```
Comment (duttonw) [inline .esbuild/plugins/qgds-plugin-copy-assets.js:21]: Only helpers (until using components folder only)

File: esbuild.js [hunk 2]
```diff
@@ -48,7 +48,11 @@ const buildConfig = {
     },
     {
       in: "./src/js/handlebars.init.js",
-      out: "./components/handlebars.init.bundle",
+      out: "./components/handlebars.init.min",
```
Comment (duttonw) [inline esbuild.js:51]: call it .min so it aligns nicely in webstorm.

File: src/components/bs5/searchInput/search.functions.js [hunk 3]
```diff
@@ -114,9 +114,9 @@ export async function showSuggestions(value = '', isDefault = false, form) {
         <div class="suggestions-category mt-2">
           <strong>Suggestions</strong>
           <ul class="mt-2">${fetchedSuggestions.slice(0, 4).map(item => {
-            const highlightedText = item.replace(new RegExp(`(${value})`, 'gi'), '<strong>$1</strong>');
-            return `<li><a href="#">${highlightedText}</a></li>`;
-          }).join('')}</ul>
+    const highlightedText = item.replace(new RegExp(`(${value})`, 'gi'), '<strong>$1</strong>');
```
Comment (duttonw) [inline src/components/bs5/searchInput/search.functions.js:117]: lint

File: src/js/handlebars.helpers.js [hunk 4]
```diff
@@ -90,15 +90,3 @@ export default function handlebarsHelpers(handlebars) {
     return handlebars.helpers.formatDate(dateToFormat);
   });
 }
-
```
Comment (duttonw) [inline src/js/handlebars.helpers.js:93]: moved to init

File: src/js/handlebars.init.js [hunk 5]
```diff
@@ -15,19 +17,18 @@ export default function init(handlebars = Handlebars) {
...
-    if (typeof handlebarsHelpers.registeredHandlebarsHelpers === 'undefined') {
-      handlebarsHelpers(handlebars);
-      handlebarsHelpers.registeredHandlebarsHelpers = true;
-    }
-    //only load once
-    if (typeof handlebarsPartials.registeredHandlebarsPartials === 'undefined') {
-      handlebarsPartials(handlebars);
-      handlebarsPartials.registeredHandlebarsPartials = true;
-      console.log('partials loaded');
+    if (!isHandlebarsHelpersAndPartialsRegistered) {
```
Comment (duttonw) [inline src/js/handlebars.init.js:20]: provide logging if loaded more than once.

File: src/js/handlebars.init.js [hunk 6]
```diff
@@ -15,19 +17,18 @@ export default function init(handlebars = Handlebars) {
...
     }
+    handlebarsHelpers(handlebars);
+
+    handlebarsPartials(handlebars);
   } else {
     console.log("Handlebars not found, init failed");
   }
 }
 
+init();
```
Comment (duttonw) [inline src/js/handlebars.init.js:33]: self init

**Review by MitchAnderson93 — APPROVED** [review]

**Review by stvp-qld — APPROVED** [review]

**Pull request merged**